src/EnergyPlus/ScheduleManager.hh

```diff
@@ -328,6 +331,109 @@ namespace ScheduleManager {
 
     int GetNumberOfSchedules(EnergyPlusData &state);
 
+    struct CSVRow
+    {
+        std::string_view operator[](std::size_t index) const;
+        std::size_t rowEnd();
+        std::string delimiter{'*'}; // eg. comma in CSV
+        void readNextRow(std::istream &str);
+
+    private:
+        std::string m_line;
+        std::vector<int> m_data;
+    };
+
+    std::istream &operator>>(std::istream &str, CSVRow &data);
+
+    void PreProcessIDF(EnergyPlus::EnergyPlusData &state, int &SchNum, int NumCommaFileSchedules);
+
+    struct schedInputIdfPreprocessObject
```

Comment:
Objects of this struct will hold the IDF file info about a particular Schedule:File that is read by the getObjectItem function. Implementing it like this hoping to bring a more object-oriented approach to this module.